### PR TITLE
Create the Environment View

### PR DESCRIPTION
**Pull request by udayasri**
- Remove loading the project from Storyboard, use  pragmatically created views instead
- Create the environmentViewController with layout
- Create the RBM specific theme
- Add UIView anchor extension
- Add UIViewController Alert extension

### CODE REVIEW

File: ExposurePlayback.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -942,10 +975,54 @@
 			path = Playable;
 			sourceTree = "<group>";
 		};
+		785B37F421A301D900FB7C7B /* ViewControllers */ = {
+			isa = PBXGroup;
+			children = (
+				785B382021A59CB700FB7C7B /* EnvironmentViewController.swift */,
+				785B37F821A3020C00FB7C7B /* LoginViewController.swift */,
```
Comment (FredrikSjoberg) [inline ExposurePlayback.xcodeproj/project.pbxproj]: LoginViewController.swift
AssetListTableViewController.swift
NewPlayerViewController.swift

are not present in the project

File: ExposurePlayback.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -942,10 +975,54 @@
...
+				780EFEED21A6D529003B2AE8 /* NewPlayerViewController.swift */,
+			);
+			path = ViewControllers;
+			sourceTree = "<group>";
+		};
+		785B380421A41A9A00FB7C7B /* Utilities */ = {
+			isa = PBXGroup;
+			children = (
+				785B382221A59DB900FB7C7B /* Theme */,
+				785B381C21A5910400FB7C7B /* TableViewDataSource.swift */,
```
Comment (FredrikSjoberg) [inline ExposurePlayback.xcodeproj/project.pbxproj]: Same with

TableViewDataSource.swift
StorageProvider.swift

File: ExposurePlayback.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -942,10 +975,54 @@
...
+		};
 		9B17CDFA2189EC13009C95C4 /* Extension */ = {
 			isa = PBXGroup;
 			children = (
 				9B17CDFB2189EC4D009C95C4 /* UIImageView+constraint.swift */,
+				785B37F221A301A900FB7C7B /* UIView+Anchor.swift */,
+				785B380221A4135A00FB7C7B /* UIView+Keyboard.swift */,
+				785B380921A41CC900FB7C7B /* UIColor+hexColor.swift */,
+				785B380D21A5481900FB7C7B /* UIViewController+Alert.swift */,
+				785B381121A554C300FB7C7B /* UITableView+EmptyMessage.swift */,
```
Comment (FredrikSjoberg) [inline ExposurePlayback.xcodeproj/project.pbxproj]: and 
UITableView+EmptyMessage.swift

File: RefApp/Navigation/MainNavigationController.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+class MainNavigationController: UINavigationController {
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.navigationBar.tintColor = ColorState.active.textFieldPlaceholder
+        self.navigationBar.barTintColor = ColorState.active.background
+        self.navigationBar.titleTextAttributes = [NSAttributedString.Key.foregroundColor: ColorState.active.textFieldPlaceholder]
+        
+        if StorageProvider.storedSessionToken != nil {
```
Comment (FredrikSjoberg) [inline RefApp/Navigation/MainNavigationController.swift]: Remove or comment away until implementation is present

File: RefApp/ViewControllers/EnvironmentViewController.swift [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                    (alert: UIAlertAction!) -> Void in
+                })
+                
+                self.popupAlert(title:NSLocalizedString("Sorry", comment: "") , message: NSLocalizedString("Please fill all fields", comment: ""), actions: [okAction], preferedStyle: .alert)
+                return
+        }
+        
+        let environment = Environment(baseUrl: environmentUrl, customer: customerName, businessUnit: businessUnit)
+        
+        let loginViewController = LoginViewController()
```
Comment (FredrikSjoberg) [inline RefApp/ViewControllers/EnvironmentViewController.swift]: Remove or comment away until implementation is present

File: RefApp/AppDelegate.swift [hunk 6]
```diff
@@ -14,13 +14,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        if let image = UIImage(named: "redbee") {
+        
+        
+        /* if let image = UIImage(named: "redbee") {
```
Comment (FredrikSjoberg) [inline RefApp/AppDelegate.swift]: Lets clean up the unused code :)

File: RefApp/ViewControllers/EnvironmentViewController.swift [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    
+    let logoImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: "redbee")
+        imageView.contentMode = .scaleAspectFill
+        return imageView
+    }()
+    
+    let environmentUrlTextField: RBMTextField = {
+        let textfield = RBMTextField(placeHolderText: "Exp URL")
```
Comment (FredrikSjoberg) [inline RefApp/ViewControllers/EnvironmentViewController.swift]: Nothing major but maybe a better placeholder that explains the url:

"Exposure base URL"?

File: RefApp/ViewControllers/EnvironmentViewController.swift [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(dissmissKeyboard))
+        view.addGestureRecognizer(tapGesture)
+        
+        view.bindToKeyboard()
+        
+        self.navigationItem.title = NSLocalizedString("Environment", comment: "")
+        view.backgroundColor = ColorState.active.background
+    }
+    
+    @objc func dissmissKeyboard() {
```
Comment (FredrikSjoberg) [inline RefApp/ViewControllers/EnvironmentViewController.swift]: typo:

dissmissKeyboard -> dismissKeyboard

File: ExposurePlayback.xcodeproj/project.pbxproj (hunk 1 above)
Comment (udayasri) [reply ExposurePlayback.xcodeproj/project.pbxproj]: noted

File: ExposurePlayback.xcodeproj/project.pbxproj (hunk 2 above)
Comment (udayasri) [reply ExposurePlayback.xcodeproj/project.pbxproj]: noted

File: ExposurePlayback.xcodeproj/project.pbxproj (hunk 3 above)
Comment (udayasri) [reply ExposurePlayback.xcodeproj/project.pbxproj]: noted

File: RefApp/Navigation/MainNavigationController.swift (hunk 4 above)
Comment (udayasri) [reply RefApp/Navigation/MainNavigationController.swift]: noted

File: RefApp/ViewControllers/EnvironmentViewController.swift (hunk 5 above)
Comment (udayasri) [reply RefApp/ViewControllers/EnvironmentViewController.swift]: noted

File: RefApp/AppDelegate.swift (hunk 6 above)
Comment (udayasri) [reply RefApp/AppDelegate.swift]: It was the previous implementation, will remove it

File: RefApp/ViewControllers/EnvironmentViewController.swift (hunk 7 above)
Comment (udayasri) [reply RefApp/ViewControllers/EnvironmentViewController.swift]: Yes, let changed it to Exposure base URL

File: RefApp/ViewControllers/EnvironmentViewController.swift (hunk 8 above)
Comment (udayasri) [reply RefApp/ViewControllers/EnvironmentViewController.swift]: noted

**Pull request merged**